lib/schema/ports.go

```diff
@@ -0,0 +1,76 @@
+/*
+Copyright 2020 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package schema
+
+// DefaultPortRanges defines the list of default ports for the cluster
+var DefaultPortRanges = PortRanges{
+	Kubernetes: []PortRange{
+		{Protocol: "tcp", From: 2379, To: 2380, Description: "etcd"},
+		{Protocol: "tcp", From: 6443, To: 6443, Description: "kubernetes API server"},
+		{Protocol: "tcp", From: 10248, To: 10255, Description: "kubernetes internal services range"},
+		{Protocol: "tcp", From: 7496, To: 7496, Description: "serf (health check agents) peer to peer"},
+		{Protocol: "tcp", From: 7373, To: 7373, Description: "serf (health check agents) peer to peer"},
+	},
+	Installer: []PortRange{
+		{Protocol: "tcp", From: 61009, To: 61010, Description: "installer ports"},
+		{Protocol: "tcp", From: 61022, To: 61025, Description: "installer ports"},
+		{Protocol: "tcp", From: 61080, To: 61080, Description: "installer ports"},
+		{Protocol: "tcp", From: 3012, To: 3012, Description: "gravity agent RPC"},
+	},
+	Vxlan: PortRange{
+		Protocol: "udp", From: 8472, To: 8472, Description: "overlay network",
+	},
+	Generic: []PortRange{
+		{Protocol: "tcp", From: 3022, To: 3025, Description: "teleport internal SSH control panel"},
+		{Protocol: "tcp", From: 3080, To: 3080, Description: "teleport Web UI"},
+		{Protocol: "tcp", From: 3008, To: 3011, Description: "internal gravity services"},
+		{Protocol: "tcp", From: 32009, To: 32009, Description: "gravity Hub control panel"},
+		{Protocol: "tcp", From: 7575, To: 7575, Description: "planet agent RPC"},
+	},
+	Reserved: []PortRange{
+		// Defines as kubernetes_port_t by default
+		{Protocol: "tcp", From: 4001, To: 4001, Description: "etcd"},
+		// Defines as afs3_callback_port_t by default
+		{Protocol: "tcp", From: 7001, To: 7001, Description: "etcd"},
+		// Defines as commplex_main_port_t by default
```

Comment:
Nit: Is "commplex" a typo (for "complex") or not? Also, should it say "Define**d**" here and above?

Reply:
No, [it's](https://github.com/fedora-selinux/selinux-policy/blob/rawhide/policy/modules/kernel/corenetwork.te.in#L138) not a typo.